### PR TITLE
Fix erb executable

### PR DESCRIPTION
**Pull request by larskanis**
This adds erb exectuable to erb's gemspec. It wasn't copied in "make install" since commit 8c97883b738ad9749848d9a10fce87df0f9b1bf3 .

Also run ERb processing unconditionally since `__FILE__` and `$0` doesn't match in a gem context.

I noticed it in RubyInstaller-CI [here](https://ci.appveyor.com/project/larskanis/rubyinstaller2-hbuor/builds/34808124/job/a331hmwi8ffp7frw#L26058). The [patched](https://github.com/oneclick/rubyinstaller2-packages/commit/7a35a9c85c30496797a5c7202367da4c6e1b4a11) version succeeds.

Error is:
```
$ erb
C:/Ruby28-x64/lib/ruby/2.8.0/rubygems.rb:277:in `find_spec_for_exe': can't find gem erb (>= 0.a) with executable erb (Gem::GemNotFoundException)
        from C:/Ruby28-x64/lib/ruby/2.8.0/rubygems.rb:296:in `activate_bin_path'
        from C:/Ruby28-x64/bin/erb.cmd:31:in `<main>'
```

### CODE REVIEW

File: lib/erb.gemspec [hunk 1]
```diff
@@ -16,7 +16,7 @@ Gem::Specification.new do |spec|
   spec.files         = Dir.chdir(File.expand_path("..", __FILE__)) do
     `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
   end
-  spec.bindir        = "exe"
-  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+  spec.bindir        = "libexe"
```
Comment (k0kubun) [inline lib/erb.gemspec:19]:
```suggestion
  spec.bindir        = "libexec"
```
Isn't this a typo?

File: libexec/erb [hunk 2]
```diff
@@ -171,6 +171,4 @@ EOU
   end
 end
 
-if __FILE__ == $0
```
Comment (k0kubun) [inline libexec/erb:174]: Why did you remove this?

File: lib/erb.gemspec (hunk 1 above)
Comment (larskanis) [reply lib/erb.gemspec]: It is. Actually any name works, but my intention was indeed "libexec".

File: libexec/erb (hunk 2 above)
Comment (larskanis) [reply libexec/erb:174]: The file was previously located in `bin/erb`, so that `__FILE__` and `$0` did match. Since https://github.com/ruby/ruby/commit/8c97883b738ad9749848d9a10fce87df0f9b1bf3 `bin/erb` is a binstub which loads `libexec/erb` from the gem directory, so that `__FILE__` and `$0` are different. Running `erb` just requires the code otherwise but doesn't execute it.

Comment (k0kubun) [reply libexec/erb:174]: Thanks, that sounds legit. I'll wait for @hsbt's reaction though, because he made the libexec change.

**Comment by hsbt** [comment]
I fixed this at https://github.com/ruby/ruby/commit/e6ac1fb6121d6cb694d107f205713d92b7b4a427. It works fine for me.

We need to keep for the standalone gem with gemspec. libexec is not provided in standalone gem.

File: libexec/erb (hunk 2 above)
Comment (hsbt) [reply libexec/erb:174]: @larskanis Can you separate this change to the another pull-request from this?

Comment (hsbt) [reply libexec/erb:174]: or rebase from the current master branch.

Comment (larskanis) [reply libexec/erb:174]: @hsbt Thanks for the fix! Rebased.

**Review by hsbt — APPROVED** [review]
I will merge this after CI passed.

**Comment by hsbt** [comment]
@larskanis Thanks! It's helpful for erb gem.
